Source/Moya+Internal.swift

```diff
@@ -228,6 +228,18 @@ private extension MoyaProvider {
             if let dataRequest = dataRequest as? T {
                 progressAlamoRequest = dataRequest
             }
+        } else if var dataRequest = progressAlamoRequest as? DownloadRequest {
+            dataRequest = dataRequest.response(queue: queue, completionHandler: { handler in
+                let result = convertResponseToResult(handler.response, request: handler.request, data: nil, error: handler.error)
+                // Inform all plugins about the response
+                plugins.forEach { $0.didReceiveResponse(result, target: target) }
+                progressCompletion?(ProgressResponse(response: result.value))
+                completion(result)
+            })
+            
+            if let dataRequest = dataRequest as? T {
+                progressAlamoRequest = dataRequest
+            }
```

Comment:
For example, we could create a protocol with the required methods from both `DataRequest` and `Download Request`, and then extend those types to conform to that protocol. Then our conditional `if let` could cast to the protocol. What do you think?